### PR TITLE
overflow intrinc can only be an argument of extractvalue instruction

### PR DESCRIPTION
**Pull request by jubitaneja**

### CODE REVIEW

**Comment by regehr** [comment]
wouldn't it be easier to do everything in one place at Parser.cpp:507?

**Comment by jubitaneja** [comment]
Oops! I need to handle this for phi instruction as well. Updating my patch now.

File: lib/Parser/Parser.cpp [hunk 1]
```diff
@@ -1132,6 +1159,23 @@ bool Parser::parseLine(std::string &ErrStr) {
...
+          case Inst::SAddWithOverflow:
+          case Inst::UAddWithOverflow:
+          case Inst::SSubWithOverflow:
+          case Inst::USubWithOverflow:
+          case Inst::SMulWithOverflow:
+          case Inst::UMulWithOverflow:
+            OverflowFlag = true;
+            break;
+          default:
+            OverflowFlag = false;
```
Comment (regehr) [inline lib/Parser/Parser.cpp]: this is already false so I don't think you need to set it

**Comment by jubitaneja** [comment]
Updated. Thanks.

**Comment by regehr** [comment]
this is too many repetitions of the same case statement, the overflow check needs to be factored out into a separate function

**Comment by regehr** [comment]
also if there's only one case you don't really want a switch

**Comment by jubitaneja** [comment]
Sure, I will fix the code. Thanks.

File: lib/Parser/Parser.cpp [hunk 2]
```diff
@@ -386,6 +387,14 @@ Inst *Parser::parseInst(std::string &ErrStr) {
   }
 }
 
+bool Parser::isOverflow(Inst::Kind IK) {
+  if (IK == Inst::SAddWithOverflow || IK == Inst::UAddWithOverflow ||
```
Comment (regehr) [inline lib/Parser/Parser.cpp]: you don't need an "if" in this kind of function, just return the result of the or-expression

File: lib/Parser/Parser.cpp [hunk 3]
```diff
@@ -576,10 +592,19 @@ bool Parser::typeCheckInst(Inst::Kind IK, unsigned &Width,
...
   // ExtractValue instruction is an index value. We don't type check
   // the operands width as the two elements vary in width.
-  if (IK != Inst::ExtractValue)
+  if (IK != Inst::ExtractValue) {
     if (!typeCheckOpsMatchingWidths(OpsMatchingWidths, ErrStr))
       return false;
 
+    for (auto Op : Ops) {
+      if (isOverflow(Op->K)) {
+        ErrStr = std::string("overflow intrinsic cannot be an operand of ") + Inst::getKindName(IK) +
```
Comment (regehr) [inline lib/Parser/Parser.cpp]: I don't think both of these explicit casts to std::string are needed

Comment (jubitaneja) [reply lib/Parser/Parser.cpp]: It does give build error if I don't typecast explicitly. For the other one, which doesn't combine multiple strings, I don't need typecast. I have fixed that.

**Comment by regehr** [comment]
LGTM

**Pull request merged**